File: eTuristickaAgencija.Service/Database/Rezervacija.cs
namespace eTuristickaAgencija.Service.Database;

public partial class Rezervacija
{
    public int Id { get; set; }
    public decimal Cijena { get; set; }
    public int? HotelId { get; set; }
    public bool? Otkazana { get; set; }
    public DateTime DatumRezervacije { get; set; }
    public DateTime CheckIn { get; set; }
    public DateTime CheckOut { get; set; }
    public int BrojOsoba { get; set; }
    public string TipSobe { get; set; }
    public int? KorisnikId { get; set; }
    public virtual Hotel? Hotel { get; set; }
    public virtual Korisnik? Korisnik { get; set; }
}

File: UI/eturistickaagencija_admin/lib/models/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? id;
  double? cijena;
  int? hotelId;
  bool? otkazana;
  DateTime? datumRezervacije;
  DateTime? checkIn;
  DateTime? checkOut;
  int? brojOsoba;
  String? tipSobe;
  int? korisnikId;

  Rezervacija(
      this.id,
      this.cijena,
      this.hotelId,
      this.otkazana,
      this.datumRezervacije,
      this.checkIn,
      this.checkOut,
      this.brojOsoba,
      this.tipSobe,
      this.korisnikId);

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}
